kubectl-cloudflow/config/config_test.go

```diff
@@ -311,6 +311,52 @@ func Test_validateConfig(t *testing.T) {
 	`)
 	assert.NotEmpty(t, validateConfig(unknownStreamletConfigSection, spec))
 
+	labelConfigSection := newConfig(`
+     cloudflow.streamlets.my-streamlet.kubernetes.pods.pod {
+		 	 labels {
+		 	 	key1 = value1
+		 	 	key2 = value2
+		 	 }
+			 containers.container {
+			 	resources {
+		            requests {
+		              cpu = 2
+		              memory = "512M"
+		            }
+		            limits {
+		              memory = "1024M"
+		            }
+		        }
+			 }
+	}
+	`)
+
+	assert.Empty(t, validateConfig(labelConfigSection, spec))
+
+	badLabelConfigSectionEmpty := newConfig(`
+     cloudflow.streamlets.my-streamlet.kubernetes.pods.pod {
+		 	 labels {
+		 	 	key1
+		 	 }
+			 containers.container {
+			 	resources {
+		            requests {
+		              cpu = 2
+		              memory = "512M"
+		            }
+		            limits {
+		              memory = "1024M"
+		            }
+		        }
+			 }
+	}
+	`)
+	assert.NotEmpty(t, validateConfig(badLabelConfigSectionEmpty, spec))
+	//The error we get is not the correct one but https://github.com/go-akka/configuration
+	// lacks the capabilities to find the config above shouldn't pass basic parsing. It should get the following error:
+	//String: 4: Key 'key1' may not be followed by token: '}' (if you intended '}' to be part of a key or string value, try enclosing the key or value in double quotes)
```

Comment:
You could try to get the value for every key found in the labels section. Does it then return empty for a key like that or does it fail?

Reply:
Fails miserably. It appends as the value of that key the rest of the HOCON file. The library we are using to read HOCON has this limitation. The error we'll get out of this won't be as useful as the rest but it actually points out in the right direction. I think the user should get some responsibility here and realize that is posting an ill-formatted file.

Reply:
Yes, this can be improved over time. At least not something you need to fix in this PR.

Reply:
Improved in this PR after all